pkg/archive/path_unix.go

```diff
@@ -2,8 +2,57 @@
 
 package archive
 
+import (
+	"fmt"
+	"io/fs"
+
+	"github.com/pkg/errors"
+	"golang.org/x/sys/unix"
+)
+
 // checkSystemDriveAndRemoveDriveLetter is the non-Windows implementation
 // of CheckSystemDriveAndRemoveDriveLetter
 func checkSystemDriveAndRemoveDriveLetter(path string) (string, error) {
 	return path, nil
 }
+
+// lgetxattr retrieves the value of the extended attribute identified by attr
+// and associated with the given path in the file system.
+// It returns a nil slice and nil error if the xattr is not set.
+func lgetxattr(path string, attr string) ([]byte, error) {
+	// Start with a 128 length byte array
+	dest := make([]byte, 128)
+	sz, err := unix.Lgetxattr(path, attr, dest)
+
+	for errors.Is(err, unix.ERANGE) {
+		// Buffer too small, use zero-sized buffer to get the actual size
+		sz, err = unix.Lgetxattr(path, attr, []byte{})
+		if err != nil {
+			return nil, wrapPathError("lgetxattr", path, attr, err)
+		}
+		dest = make([]byte, sz)
+		sz, err = unix.Lgetxattr(path, attr, dest)
+	}
+
+	if err != nil {
+		if errors.Is(err, unix.ENODATA) {
+			return nil, nil
+		}
```

Comment:
Here as well; looks like this doesn't support (Free)BSD; https://github.com/moby/buildkit/pull/5602

```
49.21 vendor/github.com/docker/docker/pkg/archive/path_unix.go:38:26: undefined: unix.ENODATA
```

Reply:
Looks like FreeBSD has `ENOATTR` instead:

> If the named attribute does not exist, or the process has no access to this attribute, errno is set to ENOATTR.


>   87 ENOATTR Attribute not	found.	The specified extended attribute  does not exist.